### DISCUSSION TITLE
[meteor] some feedbacks

### DISCUSSION THREAD

**Discussion by soolidtech** [DefinitelyTyped/DefinitelyTyped#59585] [Issues with a @types package]
Hi !
First of all, thanks for all of you for making it possible to work with Meteor & Typescript.
Here are some feedbacks from my current project:

- if you follow following [guidlines](https://guide.meteor.com/security.html#csp) from Meteor about Content Security Policy, you'll raise following typescripts errors : 
  - [ ] `WebApp.clientPrograms[arch].versionReplaceable(),`: Property 'versionReplaceable' does not exist
  - [ ] `WebApp.clientPrograms[arch].versionNonRefreshable(),`: Property 'versionNonRefreshable' does not exist
  - [ ] `WebApp.clientPrograms[arch].versionRefreshable()`: Property 'versionRefreshable' does not exist on type
  - [ ] `WebApp.clientPrograms[arch].version()`: This expression is not callable. Type 'String' has no call signatures.
  - [ ] `__meteor_runtime_config__.versions[arch]`: Cannot find name '__meteor_runtime_config__'
  - [ ] `import { Autoupdate } from 'meteor/autoupdate';` Cannot find module 'meteor/autoupdate' or its corresponding type declarations.

- If you work on i18n project, you'll certainly need to define `html` tag's `lang` property. You can do that with ` WebApp.addHtmlAttributeHook` but it raises following error : 
  - [ ]  ` WebApp.addHtmlAttributeHook`: Property 'addHtmlAttributeHook' does not exist on type 'typeof WebApp'

- If you use Assets api : 
  - [ ] `Assets.getText`: Cannot find name 'Assets'.

- If you work on a SSR project, you'll maybe want to use `RoutePolicy`, but :
  - [ ] `import { RoutePolicy } from 'meteor/routepolicy';`: Cannot find module 'meteor/routepolicy' or its corresponding type declarations.

- If you want to log stuff using dedicated meteor's tool, you'll get this error : 
  - [ ] `import { Log } from 'meteor/logging';`:  Cannot find module 'meteor/logging' or its corresponding type declarations.

That's it, I'll update this list if I find other missing types !
See you

[a comment by an automated account is not shown]

**Comment by radekmie** [comment]
Thanks for the report @soolidtech! Will you try to improve that on your own by any chance?

**Reply by soolidtech** [reply]
I'll try my best & try to find time to do this.
Be assured that this list isn't a way to ask someone else to do this, but only not lose observations that are now to be forgotten under a // @ts-ignore 😅
Have a nice day !

**Reply by radekmie** [reply]
OK, great!
